src/main/kotlin/no/nav/sykdig/digitalisering/api/DigitaliseringsoppgaveDataFetcher.kt

```diff
@@ -6,37 +6,34 @@ import com.netflix.graphql.dgs.DgsQuery
 import com.netflix.graphql.dgs.InputArgument
 import graphql.schema.DataFetchingEnvironment
 import no.nav.sykdig.digitalisering.exceptions.ClientException
-import no.nav.sykdig.digitalisering.pdl.PersonService
+import no.nav.sykdig.digitalisering.model.FerdistilltRegisterOppgaveValues
+import no.nav.sykdig.digitalisering.model.RegisterOppgaveValues
+import no.nav.sykdig.digitalisering.model.UferdigRegisterOppgaveValues
 import no.nav.sykdig.generated.DgsConstants
-import no.nav.sykdig.generated.types.DiagnoseInput
 import no.nav.sykdig.generated.types.Digitaliseringsoppgave
-import no.nav.sykdig.generated.types.PeriodeInput
 import no.nav.sykdig.generated.types.SykmeldingUnderArbeidStatus
 import no.nav.sykdig.generated.types.SykmeldingUnderArbeidValues
 import no.nav.sykdig.logger
 import no.nav.sykdig.utils.toOffsetDateTimeAtNoon
-import java.time.OffsetDateTime
+import org.springframework.security.access.prepost.PreAuthorize
 import kotlin.contracts.ExperimentalContracts
 import kotlin.contracts.contract
 
 @DgsComponent
-class OppgaveDataFetcher(
-    private val oppgaveService: OppgaveService,
-    private val personService: PersonService,
+class DigitaliseringsoppgaveDataFetcher(
+    val digitaliseringsoppgaveService: DigitaliseringsoppgaveService
 ) {
-    val log = logger()
+    companion object {
+        private val log = logger()
+    }
 
+    @PreAuthorize("@oppgaveSecurityService.hasAccessToOppgave(#oppgaveId)")
     @DgsQuery(field = DgsConstants.QUERY.Oppgave)
     fun getOppgave(@InputArgument oppgaveId: String): Digitaliseringsoppgave {
-        val oppgave = oppgaveService.getOppgave(oppgaveId)
-        val person = personService.hentPerson(
-            fnr = oppgave.fnr,
-            sykmeldingId = oppgave.sykmeldingId.toString()
-        )
-
-        return mapToDigitaliseringsoppgave(oppgave, person)
+        return mapToDigitaliseringsoppgave(digitaliseringsoppgaveService.getDigitaiseringsoppgave(oppgaveId))
     }
 
+    @PreAuthorize("@oppgaveSecurityService.hasAccessToOppgave(#oppgaveId)")
```

Comment:
Magic

Reply:
Dette var skikkelig bra!! Jeg har vært litt nervøs for om tilgangskontrollen skulle bli glemt siden den skjedde litt bortgjemt i et metodekall som egentlig bare så ut til å hente oppgaven 😅 Jeg visste ikke at det gikk an å gjøre sånn som det her :)